modules/api_loader.lua

```diff
@@ -1,81 +1,146 @@
 local xml = require('xml')
 local module = { }
 
-local function get_name(xml_element)
-    local parent_name = xml_element:parent():attr("name")
-    local name = xml_element:attr("name")
-    if(module.include_parent_name) then
-        name = parent_name .. "." .. name
+-- Include result codes that are elements in functions from Mobile Api.
+-- Each function with paremeter resultCode that has type Result
+-- should contain types of Resultcode directly in function.
+-- Other resultCodes are kept in structs  
+local function LoadResultCodes( param )
+  local resultCodes ={}
+  local i = 1
+  for _, item in ipairs(param:children("element")) do
+     local name = item:attr("name")
+     resultCodes[i]=name
+     i=i + 1
     end
-    return name
+  return resultCodes
 end
 
-local function loadEnums(api, dest)
-  local enums = api:xpath("//interface/enum")
-  for _, e in ipairs(enums) do
-    local enum = { }
-    local i = 1
-    for _, item in ipairs(e:children("element")) do
-      enum[item:attr("name")] = i
-      i = i + 1
-    end
-    dest.enum[get_name(e)] = enum
+--Load parameteres in function. Load ResultCodes if 
+--type of parameter is "Result"
+local function LoadParamsInFunction(param, interface)
+  local name = param:attr("name")
+  local p_type = param:attr("type")
+  local mandatory = param:attr("mandatory")
+  local array = param:attr("array")
+
+  if mandatory == nil then 
+    mandatory = true
+  end
+
+  if array == nil then 
+    array = false
+  end
+
+  local result_codes = nil
+  if name == "resultCode" and p_type == "Result" then 
+    result_codes  = LoadResultCodes(param) 
   end
+
+  local data = {}
+  data["type"]=p_type
```

Comment:
@LevchenkoS No, because some of this field, like `p_type`, is using in `result_codes` filling